R/pmcmc.R

```diff
@@ -119,20 +119,43 @@ pmcmc <- function(pars, filter, n_steps, save_state = FALSE,
     }
   }
 
-  pars <- set_colnames(list_to_matrix(history_pars$get()),
-                       names(curr_pars))
+  pars_matrix <- set_colnames(list_to_matrix(history_pars$get()),
+                              names(curr_pars))
   probabilities <- set_colnames(list_to_matrix(history_probabilities$get()),
                        c("log_prior", "log_likelihood", "log_posterior"))
 
-  state <- trajectories <- NULL
+  predict <- state <- trajectories <- NULL
   if (save_state) {
     state <- t(list_to_matrix(history_state$get()))
+
+    ## Extract the transformation function from the pars object as
+    ## it's all we want; this is a bit of a hack but it works for now
+    ## at least.
+    transform <- pars[[".__enclos_env__"]]$private$transform
+
+    ## Information about how the particle filter was configured:
+    info <- filter$predict_info()
+
+    ## This information is required in order to run predictions but
+    ## adds a significant space overhead to the saved data (model is
+    ## fairly large, and transform could capture anything in scope if
+    ## it is a closure).
+    predict <- list(transform = transform,
+                    model = filter$model,
+                    n_threads = info$n_threads,
+                    index = info$index,
+                    rate = info$rate,
+                    step = last(info$step))
   }
   if (save_trajectories) {
-    trajectories <- list_to_array(history_trajectories$get())
+    info <- filter$predict_info()
+    trajectories_state <-
+      aperm(list_to_array(history_trajectories$get()), c(1, 3, 2))
```

Comment:
Why 1, 3, 2? Add a comment to this line perhaps

Reply:
comment added in f57c8c1 (this is an array permutation, a generalisation of a transpose)